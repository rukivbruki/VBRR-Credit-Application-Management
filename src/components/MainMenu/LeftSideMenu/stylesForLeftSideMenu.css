.el-tabs--left .el-tabs__header.is-left {
  margin-right: 0;
  width: 100%;
  background-color: #333333;
}

.el-tabs--left {
  border: 0;
  margin: 0;
}

.el-tabs--border-card > .el-tabs__content {
  padding: 0;
}

.el-tabs__nav {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.el-tabs__item {
  font-size: 16px;
}

.el-tabs--left.el-tabs--border-card .el-tabs__item.is-left {
  color: #d3dce6;
  padding-top: 2px;
  height: 50px;
}

.el-tabs--left.el-tabs--border-card .el-tabs__item.is-left:hover {
  color: #d3dce6;
  background-color: #4c4c4c;
  margin: 0;
}

.el-tabs--left.el-tabs--border-card .el-tabs__item.is-left.is-active {
  border-bottom: solid rgb(255, 211, 0);
  border-bottom-width: 5px;
  border-top: 0;
  background-color: #333333;
  margin: 0;
}
