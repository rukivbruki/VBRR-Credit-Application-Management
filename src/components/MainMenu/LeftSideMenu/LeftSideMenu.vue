<template>
  <el-aside>
    <el-tabs
      :tab-position="tabPosition"
      :stretch="true"
      type="border-card"
      @tab-click="tabClick"
    >
      <el-tab-pane label="Действия, пользователи, процессы" />
      <el-tab-pane label="Представления, данные, файлы" />
      <el-tab-pane label="Прочее, обслуживание, субъекты" />
      <el-tab-pane label="Кредитование, досье" />
      <el-tab-pane label="Нумерация договора" />
    </el-tabs>
  </el-aside>
</template>

<script>
import { defineComponent } from 'vue';
import './stylesForLeftSideMenu.css';

export default defineComponent({
  mounted() {
    this.clickedTabIndex = sessionStorage.getItem('clickedTabIndex');

    setTimeout(() => {
      let curTab = document.getElementById('tab-' + this.clickedTabIndex);

      this.clickedTabIndex && this.unselectTab();
      curTab.setAttribute('aria-selected', 'true');
      curTab.classList.add('is-active');
    }, 0);
  },
  methods: {
    tabClick(targetName) {
      this.$store.dispatch('changeCardsToShow', targetName.index);
      this.clickedTabIndex = targetName.index;
      sessionStorage.setItem('clickedTabIndex', targetName.index);
    },
    unselectTab() {
      const prevTab = document.getElementById('tab-0');

      prevTab.setAttribute('aria-selected', 'false');
      prevTab.classList.remove('is-active');
    },
  },
  data() {
    return {
      tabPosition: 'left',
      clickedTabIndex: 0,
    };
  },
});
</script>

<style scoped>
.el-aside {
  display: flex;
  align-content: center;
  --el-aside-width: 100%;
  color: #333;
  text-align: center;
  line-height: 1.5;
  background-color: #d3dce6;
}

.el-tabs {
  width: 100%;
}
</style>
