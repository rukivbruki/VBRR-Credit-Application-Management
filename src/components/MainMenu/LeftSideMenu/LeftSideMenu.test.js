import { shallowMount } from '@vue/test-utils';
import LeftSideMenu from './LeftSideMenu';
describe('LeftSideMenu.vue render', () => {
  test('renders correctly', () => {
    expect(shallowMount(LeftSideMenu).element).toMatchSnapshot();
  });
});
