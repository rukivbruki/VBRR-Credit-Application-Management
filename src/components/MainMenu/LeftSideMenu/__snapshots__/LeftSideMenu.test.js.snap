// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftSideMenu.vue render renders correctly 1`] = `
<el-aside>
  <el-tabs
    stretch="true"
    tab-position="left"
    type="border-card"
  >
    <el-tab-pane
      label="Действия, пользователи, процессы"
    />
    <el-tab-pane
      label="Представления, данные, файлы"
    />
    <el-tab-pane
      label="Прочее, обслуживание, субъекты"
    />
    <el-tab-pane
      label="Кредитование, досье"
    />
    <el-tab-pane
      label="Нумерация договора"
    />
  </el-tabs>
</el-aside>
`;
